typesummary/app/src/main/java/typesummary/App.java

```diff
@@ -0,0 +1,125 @@
+                                            .map(classInfo -> classInfo.getName())
+                                            .distinct()
+                                            .collect(Collectors.toList());
+        classNames.sort(Comparator.naturalOrder());
+        return classNames;
+    }
+    private static void serializeInterfaces(final ILogWriter writer, final List<String> classNames) throws Exception {
+        for (String c : classNames) {
+            Class<?> clazz = Class.forName(c);
+            if(clazz.isInterface()) {
+                String classHeadLine = "interface " + clazz.getName();
+                Class<?> superClass = clazz.getSuperclass();
+                if(superClass != null && superClass != Object.class) {
+                    classHeadLine += " : " + superClass.getName();
+                }
+                writer.write(classHeadLine);
+                serializeFields(clazz, writer);
+                serializeMethods(clazz, writer);
+            }
+        }
+    }
+    private static void serializeClasses(final ILogWriter writer, final List<String> classNames) throws Exception {
+        for (String c : classNames) {
+            Class<?> clazz = Class.forName(c);
+            if(!clazz.isEnum() && !clazz.isInterface()) {
+                String classHeadLine = "class " + clazz.getName();
+                Class<?> superClass = clazz.getSuperclass();
+                if(superClass != null && superClass != Object.class) {
+                    classHeadLine += " : " + superClass.getName();
+                }
+                writer.write(classHeadLine);
+                serializeFields(clazz, writer);
+                serializeMethods(clazz, writer);
+            }
+        }
+    }
+    private static void serializeFields(final Class<?> clazz, final ILogWriter writer) {
+        final Field[] fields = clazz.getFields();
+        Arrays.sort(fields, (o1, o2) -> o1.getName().compareTo(o2.getName()));
+        for(Field field : fields) {
+            writer.write("property " + field.getName() + " : " + field.getType().getName(), 2);
+        }
+    }
+    private static List<String> methodsNameToSkip;
+    private static void serializeMethods(final Class<?> clazz, final ILogWriter writer) {
+        final Method[] methods = clazz.getMethods();
+        Arrays.sort(methods, (o1, o2) -> o1.getName().compareTo(o2.getName()));
+        for(Method method : methods) {
+            if(!methodsNameToSkip.contains(method.getName())) {
+                writer.write("method " + method.getName(), 2);
+                writer.write("return type " +method.getReturnType().getName(), 4);
+                serializeParameters(method, writer);
+            }
+        }
+    }
+    private static void serializeParameters(final Method method, final ILogWriter writer) {
+        for(Parameter parameter: method.getParameters()) {
+            writer.write("param " + parameter.getName() + " : " + parameter.getType().getName(), 4);
```

Comment:
Is it expected that `parameter.getName()` only returns `arg0`, `arg1` etc.?

Reply:
do you mean it's never returning the actual parameter name? I'll take a look, I think I missed that. thanks!

Reply:
yeah, typeSummary.txt has only those as parameter names.

Reply:
so this is actually a side effect caused by the fact that the SDK is technically still building targeting java 7 at this point (v3 upgrades to java 8). I changed the project structure so the typesummary tool builds the SDK itself (instead of 2 separate steps) under java 8 and now the parameters names show up.

The CI should update the summary text with the parameters names now.
some more reading https://github.com/paul-hammant/paranamer